### PR TITLE
Mapping/Documenting: net.minecraft.client.sound

### PR DESCRIPTION
**Pull request by Lanse505**

### CODE REVIEW

File: data/net/minecraft/client/sounds/SoundEngine.mapping [hunk 1]
```diff
@@ -28,46 +30,121 @@ CLASS net/minecraft/client/sounds/SoundEngine
 	METHOD destroy ()V
 		COMMENT Cleans up the Sound System
 	METHOD getVolume (Lnet/minecraft/sounds/SoundSource;)F
+		COMMENT {@return Returns the volume value pinned between 0.0f and 1.0f for a given {@code SoundSource} category}
```
Comment (Matyrobbrt) [inline data/net/minecraft/client/sounds/SoundEngine.mapping]:
```suggestion
		COMMENT {@return the volume value pinned between 0.0f and 1.0f for a given {@linkplain SoundSource} category}
```

File: data/net/minecraft/client/sounds/SoundEngine.mapping [hunk 2]
```diff
@@ -20,6 +21,7 @@ CLASS net/minecraft/client/sounds/SoundEngine
 		ARG 2 options
 		ARG 3 resourceManager
 	METHOD addEventListener (Lnet/minecraft/client/sounds/SoundEventListener;)V
+		COMMENT Adds a new {@code SoundEventListener}
```
Comment (Matyrobbrt) [inline data/net/minecraft/client/sounds/SoundEngine.mapping]: This javadoc seems useless as it stands

File: data/net/minecraft/client/sounds/SoundEngine.mapping [hunk 3]
```diff
@@ -28,46 +30,121 @@ CLASS net/minecraft/client/sounds/SoundEngine
...
 		ARG 0 sound
 	METHOD stop (Lnet/minecraft/client/resources/sounds/SoundInstance;)V
+		COMMENT Stops the provided {@code SoundInstace} from continuing playing.
 		ARG 1 sound
 	METHOD stop (Lnet/minecraft/resources/ResourceLocation;Lnet/minecraft/sounds/SoundSource;)V
 		ARG 1 soundName
 		ARG 2 category
 	METHOD stopAll ()V
-		COMMENT Stops all currently playing sounds
+		COMMENT Stops all currently playing instances of {@code SoundInstance}
```
Comment (Matyrobbrt) [inline data/net/minecraft/client/sounds/SoundEngine.mapping]: I'd say the previous javadoc was good enough

File: data/net/minecraft/client/sounds/SoundEngine.mapping [hunk 4]
```diff
@@ -28,46 +30,121 @@ CLASS net/minecraft/client/sounds/SoundEngine
...
+		COMMENT <p>
+		COMMENT Finally, it sets the device pool state to indicate that the device check is complete.
+		COMMENT
+		COMMENT @return {@code true} if a change in the audio device is needed, {@code false} otherwise.
 	METHOD shouldLoopAutomatically (Lnet/minecraft/client/resources/sounds/SoundInstance;)Z
 		ARG 0 sound
 	METHOD shouldLoopManually (Lnet/minecraft/client/resources/sounds/SoundInstance;)Z
 		ARG 0 sound
 	METHOD stop (Lnet/minecraft/client/resources/sounds/SoundInstance;)V
+		COMMENT Stops the provided {@code SoundInstace} from continuing playing.
```
Comment (Matyrobbrt) [inline data/net/minecraft/client/sounds/SoundEngine.mapping]:
```suggestion
		COMMENT Stops the provided {@linkplain SoundInstace} from continuing to play. 
```

File: data/net/minecraft/client/sounds/SoundEngine.mapping [hunk 5]
```diff
@@ -28,46 +30,121 @@ CLASS net/minecraft/client/sounds/SoundEngine
...
 		ARG 2 delay
 	METHOD queueTickingSound (Lnet/minecraft/client/resources/sounds/TickableSoundInstance;)V
 		ARG 1 tickableSound
+	METHOD reload ()V
+		COMMENT Reloads the sound engine.
+		COMMENT <p>
+		COMMENT This method clears the warning set, checks for missing sound events, destroys the current sound system,
+		COMMENT and reloads the library.
 	METHOD removeEventListener (Lnet/minecraft/client/sounds/SoundEventListener;)V
+		COMMENT Removes a specific {@code SoundEventListener}
```
Comment (Matyrobbrt) [inline data/net/minecraft/client/sounds/SoundEngine.mapping]: This javadoc seems useless as it stands

**Review by Matyrobbrt — CHANGES_REQUESTED** [review]

File: data/net/minecraft/client/sounds/ChannelAccess.mapping [hunk 6]
```diff
@@ -1,13 +1,23 @@
...
+			COMMENT the consumer to execute on the stream of channels
 	CLASS ChannelHandle
+		COMMENT Represents a handle to a channel.
 		METHOD <init> (Lnet/minecraft/client/sounds/ChannelAccess;Lcom/mojang/blaze3d/audio/Channel;)V
 			ARG 2 channel
 		METHOD execute (Ljava/util/function/Consumer;)V
 			ARG 1 soundConsumer
+		METHOD isStopped ()Z
+			COMMENT Returns whether the channel has been stopped.
+			COMMENT @return {@code true} if the channel has been stopped, {@code false} otherwise
```
Comment (Matyrobbrt) [inline data/net/minecraft/client/sounds/ChannelAccess.mapping]: Can be compressed into an inline return,

File: data/net/minecraft/client/sounds/MusicManager.mapping [hunk 7]
```diff
@@ -1,9 +1,26 @@
...
+	FIELD STARTING_DELAY I
+		COMMENT The delay before starting to play the next song.
+	FIELD nextSongDelay I
+		COMMENT The delay until the next song starts.
 	METHOD <init> (Lnet/minecraft/client/Minecraft;)V
 		ARG 1 minecraft
 	METHOD isPlayingMusic (Lnet/minecraft/sounds/Music;)Z
+		COMMENT Returns whether the specified Music selector is currently playing.
+		COMMENT <p>
+		COMMENT @return true if the Music selector is currently playing, false otherwise
```
Comment (Matyrobbrt) [inline data/net/minecraft/client/sounds/MusicManager.mapping]: Can be compressed into an inline return.

File: data/net/minecraft/client/sounds/SoundBufferLibrary.mapping [hunk 8]
```diff
@@ -1,14 +1,34 @@
...
+		COMMENT The ResourceProvider used for loading sound resources.
 	METHOD <init> (Lnet/minecraft/server/packs/resources/ResourceProvider;)V
 		ARG 1 resourceManager
+	METHOD clear ()V
+		COMMENT Clears the cache of SoundBuffers.
 	METHOD getCompleteBuffer (Lnet/minecraft/resources/ResourceLocation;)Ljava/util/concurrent/CompletableFuture;
+		COMMENT Returns a CompletableFuture containing the complete SoundBuffer for the specified sound resource.
+		COMMENT The SoundBuffer is loaded asynchronously and cached for future use.
+		COMMENT <p>
+		COMMENT @return a CompletableFuture containing the complete SoundBuffer
```
Comment (Matyrobbrt) [inline data/net/minecraft/client/sounds/SoundBufferLibrary.mapping]: Can be compressed into an inline return.

File: data/net/minecraft/client/sounds/SoundBufferLibrary.mapping [hunk 9]
```diff
@@ -1,14 +1,34 @@
...
 		ARG 1 soundID
+			COMMENT the ResourceLocation of the sound
 	METHOD getStream (Lnet/minecraft/resources/ResourceLocation;Z)Ljava/util/concurrent/CompletableFuture;
+		COMMENT Returns a CompletableFuture containing the AudioStream for the specified sound resource.
+		COMMENT The AudioStream is loaded asynchronously.
+		COMMENT <p>
+		COMMENT @return a CompletableFuture containing the AudioStream
```
Comment (Matyrobbrt) [inline data/net/minecraft/client/sounds/SoundBufferLibrary.mapping]: Can be compressed into an inline return.

File: data/net/minecraft/client/sounds/SoundEngine.mapping [hunk 10]
```diff
@@ -22,52 +31,156 @@ CLASS net/minecraft/client/sounds/SoundEngine
...
 		ARG 1 sound
+			COMMENT the {@linkplain SoundInstance} being played
+	METHOD calculateVolume (FLnet/minecraft/sounds/SoundSource;)F
+		COMMENT Calculates the volume of the sound being played.
+		COMMENT <p>
+		COMMENT Clamping the sound between 0.0f and 1.0f.
 	METHOD calculateVolume (Lnet/minecraft/client/resources/sounds/SoundInstance;)F
+		COMMENT Calculates the volume for the sound being played.
+		COMMENT <p>
+		COMMENT Delegating to {@code #calculateVolume(float, SoundSource)}
```
Comment (Matyrobbrt) [inline data/net/minecraft/client/sounds/SoundEngine.mapping]:
```suggestion
		COMMENT Delegates to {@code #calculateVolume(float, SoundSource)}
```

File: data/net/minecraft/client/sounds/SoundEngine.mapping [hunk 11]
```diff
@@ -22,52 +31,156 @@ CLASS net/minecraft/client/sounds/SoundEngine
...
 	METHOD calculatePitch (Lnet/minecraft/client/resources/sounds/SoundInstance;)F
+		COMMENT Calculates the pitch of the sound being played.
+		COMMENT <p>
+		COMMENT Clamping the sound between 0.5f and 2.0f.
 		ARG 1 sound
+			COMMENT the {@linkplain SoundInstance} being played
+	METHOD calculateVolume (FLnet/minecraft/sounds/SoundSource;)F
+		COMMENT Calculates the volume of the sound being played.
+		COMMENT <p>
+		COMMENT Clamping the sound between 0.0f and 1.0f.
```
Comment (Matyrobbrt) [inline data/net/minecraft/client/sounds/SoundEngine.mapping]:
```suggestion
		COMMENT Clamps the sound between 0.0f and 1.0f.
```

File: data/net/minecraft/client/sounds/SoundEngine.mapping [hunk 12]
```diff
@@ -22,52 +31,156 @@ CLASS net/minecraft/client/sounds/SoundEngine
 	METHOD addEventListener (Lnet/minecraft/client/sounds/SoundEventListener;)V
 		ARG 1 listener
 	METHOD calculatePitch (Lnet/minecraft/client/resources/sounds/SoundInstance;)F
+		COMMENT Calculates the pitch of the sound being played.
+		COMMENT <p>
+		COMMENT Clamping the sound between 0.5f and 2.0f.
```
Comment (Matyrobbrt) [inline data/net/minecraft/client/sounds/SoundEngine.mapping]:
```suggestion
		COMMENT Clamps the sound between 0.5f and 2.0f.
```

File: data/net/minecraft/client/sounds/SoundEngine.mapping [hunk 13]
```diff
@@ -22,52 +31,156 @@ CLASS net/minecraft/client/sounds/SoundEngine
...
 		ARG 1 sound
 	METHOD destroy ()V
 		COMMENT Cleans up the Sound System
 	METHOD getVolume (Lnet/minecraft/sounds/SoundSource;)F
+		COMMENT {@return the volume value pinned between 0.0f and 1.0f for a given {@linkplain SoundSource} category}
 		ARG 1 category
 	METHOD isActive (Lnet/minecraft/client/resources/sounds/SoundInstance;)Z
+		COMMENT Checks if a SoundInstance is active.
+		COMMENT <p>
+		COMMENT @return true if the SoundInstance is active, false otherwise
```
Comment (Matyrobbrt) [inline data/net/minecraft/client/sounds/SoundEngine.mapping]: Can be compressed into an inline return.

File: data/net/minecraft/client/sounds/SoundEngine.mapping [hunk 14]
```diff
@@ -22,52 +31,156 @@ CLASS net/minecraft/client/sounds/SoundEngine
...
 		ARG 1 sound
+			COMMENT The SoundInstance to check
```
Comment (Matyrobbrt) [inline data/net/minecraft/client/sounds/SoundEngine.mapping]:
```suggestion
			COMMENT the SoundInstance to check
```

File: data/net/minecraft/client/sounds/SoundEngine.mapping [hunk 15]
```diff
@@ -22,52 +31,156 @@ CLASS net/minecraft/client/sounds/SoundEngine
...
 	METHOD lambda$updateCategoryVolume$1 (Lnet/minecraft/client/resources/sounds/SoundInstance;Lnet/minecraft/client/sounds/ChannelAccess$ChannelHandle;)V
 		ARG 1 instance
 		ARG 2 source
 	METHOD loadLibrary ()V
-		COMMENT Tries to add the paulscode library and the relevant codecs. If it fails, the master volume will be set to zero.
+		COMMENT Loads the sound library if it has not been loaded already.
+		COMMENT <p>
+		COMMENT If loading is successful, the library is initialized, and the sound engine is started.
+		COMMENT <p>
+		COMMENT Otherwise, an error message is logged, and sounds and music are turned off.
```
Comment (Matyrobbrt) [inline data/net/minecraft/client/sounds/SoundEngine.mapping]: Should be the same sentence.

File: data/net/minecraft/client/sounds/SoundEngine.mapping [hunk 16]
```diff
@@ -22,52 +31,156 @@ CLASS net/minecraft/client/sounds/SoundEngine
...
 	METHOD pause ()V
 		COMMENT Pauses all currently playing sounds
 	METHOD play (Lnet/minecraft/client/resources/sounds/SoundInstance;)V
+		COMMENT Plays a given sound instance.
+		COMMENT <p>
+		COMMENT If the sound engine is not loaded or the sound instance cannot be played, the method returns early.
+		COMMENT <p>
+		COMMENT The method fulfills the following parts:
+		COMMENT <ol>
```
Comment (Matyrobbrt) [inline data/net/minecraft/client/sounds/SoundEngine.mapping]: Does this list need to be ordered?

File: data/net/minecraft/client/sounds/SoundEngine.mapping [hunk 17]
```diff
@@ -22,52 +31,156 @@ CLASS net/minecraft/client/sounds/SoundEngine
...
+		COMMENT   <li>Performs a series of checks to determine if it can play a sound</li>
+		COMMENT   <li>Handles the playing of instances of {@code SoundInstance}</li>
+		COMMENT   <li>Logs potential errors that may have occured</li>
+		COMMENT   <li>Handles mapping instances of {@code SoundInstance} to specific audio channels</li>
+		COMMENT   <li>Handles deletion times for active instances of {@code SoundInstance}</li>
+		COMMENT   <li>Calculates and handles various sound properties such as volume, pitch, attenuation, looping, position and relative, </li>
+		COMMENT </ol>
+		COMMENT <p>
+		COMMENT @implNote This method assumes proper synchronization or thread confinement mechanisms are in place.
```
Comment (Matyrobbrt) [inline data/net/minecraft/client/sounds/SoundEngine.mapping]:
```suggestion
		COMMENT @implNote This method assumes proper synchronization or that thread confinement mechanisms are in place.
```

File: data/net/minecraft/client/sounds/SoundEngine.mapping [hunk 18]
```diff
@@ -22,52 +31,156 @@ CLASS net/minecraft/client/sounds/SoundEngine
...
 		ARG 1 sound
+			COMMENT The sound instance to be played.
```
Comment (Matyrobbrt) [inline data/net/minecraft/client/sounds/SoundEngine.mapping]:
```suggestion
			COMMENT the sound instance to be played
```

File: data/net/minecraft/client/sounds/SoundEngine.mapping [hunk 19]
```diff
@@ -22,52 +31,156 @@ CLASS net/minecraft/client/sounds/SoundEngine
...
 	METHOD playDelayed (Lnet/minecraft/client/resources/sounds/SoundInstance;I)V
 		COMMENT Adds a sound to play in n tick
 		ARG 1 sound
 		ARG 2 delay
 	METHOD queueTickingSound (Lnet/minecraft/client/resources/sounds/TickableSoundInstance;)V
+		COMMENT Queues a new {@linkplain TickingCodeInstance}
 		ARG 1 tickableSound
+			COMMENT the {@linkplain TickableSoundInstance} to add
```
Comment (Matyrobbrt) [inline data/net/minecraft/client/sounds/SoundEngine.mapping]:
```suggestion
			COMMENT the {@linkplain TickableSoundInstance} to queue
```

File: data/net/minecraft/client/sounds/SoundEngine.mapping [hunk 20]
```diff
@@ -22,52 +31,156 @@ CLASS net/minecraft/client/sounds/SoundEngine
...
+	METHOD reload ()V
+		COMMENT Reloads the sound engine.
+		COMMENT <p>
+		COMMENT This method clears the warning set, checks for missing sound events, destroys the current sound system,
+		COMMENT and reloads the library.
```
Comment (Matyrobbrt) [inline data/net/minecraft/client/sounds/SoundEngine.mapping]:
```suggestion
		COMMENT This method clears the warning set, checks for missing sound events, destroys the current sound system, and reloads the library.
```

File: data/net/minecraft/client/sounds/SoundEngine.mapping [hunk 21]
```diff
@@ -22,52 +31,156 @@ CLASS net/minecraft/client/sounds/SoundEngine
...
 	METHOD removeEventListener (Lnet/minecraft/client/sounds/SoundEventListener;)V
 		ARG 1 listener
 	METHOD requestPreload (Lnet/minecraft/client/resources/sounds/Sound;)V
+		COMMENT Requests to preload a specific {@linkplain Sound} instance
```
Comment (Matyrobbrt) [inline data/net/minecraft/client/sounds/SoundEngine.mapping]:
```suggestion
		COMMENT Requests a specific {@linkplain Sound} instance to be preloaded.
```

File: data/net/minecraft/client/sounds/SoundEngine.mapping [hunk 22]
```diff
@@ -22,52 +31,156 @@ CLASS net/minecraft/client/sounds/SoundEngine
...
 		ARG 1 sound
 	METHOD requiresManualLooping (Lnet/minecraft/client/resources/sounds/SoundInstance;)Z
+		COMMENT Checks if a SoundInstance requires manual looping.
+		COMMENT <p>
+		COMMENT @return true if the SoundInstance requires manual looping, false otherwise
```
Comment (Matyrobbrt) [inline data/net/minecraft/client/sounds/SoundEngine.mapping]: Can be compressed into an inline return.

File: data/net/minecraft/client/sounds/SoundEngine.mapping [hunk 23]
```diff
@@ -22,52 +31,156 @@ CLASS net/minecraft/client/sounds/SoundEngine
...
 		ARG 0 sound
+			COMMENT The SoundInstance to check
```
Comment (Matyrobbrt) [inline data/net/minecraft/client/sounds/SoundEngine.mapping]:
```suggestion
			COMMENT the SoundInstance to check
```

File: data/net/minecraft/client/sounds/SoundEngine.mapping [hunk 24]
```diff
@@ -22,52 +31,156 @@ CLASS net/minecraft/client/sounds/SoundEngine
...
+		COMMENT <p>
+		COMMENT If the preferred sound device is an empty string and the system default audio device has changed, it logs an information message
+		COMMENT and sets the device pool state to indicate a change has been detected.
+		COMMENT <p>
+		COMMENT If the preferred sound device is not an empty string, it checks if the current device name is different from the preferred device name
+		COMMENT and if the preferred device is available in the list of available sound devices.
+		COMMENT <p>
+		COMMENT If both conditions are true, it logs an information message and sets the device pool state to indicate a change has been detected.
+		COMMENT <p>
+		COMMENT Finally, it sets the device pool state to indicate that the device check is complete.
```
Comment (Matyrobbrt) [inline data/net/minecraft/client/sounds/SoundEngine.mapping]: All of this should be rewritten in passive voice.

File: data/net/minecraft/client/sounds/SoundEngine.mapping [hunk 25]
```diff
@@ -22,52 +31,156 @@ CLASS net/minecraft/client/sounds/SoundEngine
...
+		COMMENT
+		COMMENT @return {@code true} if a change in the audio device is needed, {@code false} otherwise.
 	METHOD shouldLoopAutomatically (Lnet/minecraft/client/resources/sounds/SoundInstance;)Z
+		COMMENT Checks if a SoundInstance should loop automatically.
+		COMMENT <p>
+		COMMENT @return true if the SoundInstance should loop automatically, false otherwise
```
Comment (Matyrobbrt) [inline data/net/minecraft/client/sounds/SoundEngine.mapping]: Can be compressed into an inline return.

File: data/net/minecraft/client/sounds/SoundEngine.mapping [hunk 26]
```diff
@@ -22,52 +31,156 @@ CLASS net/minecraft/client/sounds/SoundEngine
...
 		ARG 0 sound
+			COMMENT The SoundInstance to check
```
Comment (Matyrobbrt) [inline data/net/minecraft/client/sounds/SoundEngine.mapping:127]:
```suggestion
			COMMENT the SoundInstance to check
```

File: data/net/minecraft/client/sounds/SoundEngine.mapping [hunk 27]
```diff
@@ -22,52 +31,156 @@ CLASS net/minecraft/client/sounds/SoundEngine
...
 	METHOD shouldLoopManually (Lnet/minecraft/client/resources/sounds/SoundInstance;)Z
+		COMMENT Checks if a SoundInstance should loop manually.
+		COMMENT <p>
+		COMMENT @return true if the SoundInstance should loop manually, false otherwise
```
Comment (Matyrobbrt) [inline data/net/minecraft/client/sounds/SoundEngine.mapping]: Can be compressed into an inline return.

File: data/net/minecraft/client/sounds/SoundEventListener.mapping [hunk 28]
```diff
@@ -1,4 +1,9 @@
 CLASS net/minecraft/client/sounds/SoundEventListener
+	COMMENT The SoundEventListener interface defines a listener for sound events.
+	COMMENT Classes implementing this interface can be registered as listeners to receive notifications when a sound is played.
 	METHOD onPlaySound (Lnet/minecraft/client/resources/sounds/SoundInstance;Lnet/minecraft/client/sounds/WeighedSoundEvents;)V
+		COMMENT Called when a sound is played.
 		ARG 1 sound
+			COMMENT The sound instance being played
```
Comment (Matyrobbrt) [inline data/net/minecraft/client/sounds/SoundEventListener.mapping]:
```suggestion
			COMMENT the sound instance being played
```

File: data/net/minecraft/client/sounds/SoundEventListener.mapping [hunk 29]
```diff
@@ -1,4 +1,9 @@
...
 		ARG 2 accessor
+			COMMENT The accessor for weighed sound events associated with the sound
```
Comment (Matyrobbrt) [inline data/net/minecraft/client/sounds/SoundEventListener.mapping]:
```suggestion
			COMMENT the accessor for weighed sound events associated with the sound
```

File: data/net/minecraft/client/sounds/SoundEngine.mapping [hunk 30]
```diff
@@ -22,52 +31,156 @@ CLASS net/minecraft/client/sounds/SoundEngine
...
 		ARG 0 sound
+			COMMENT The SoundInstance to check
```
Comment (Matyrobbrt) [inline data/net/minecraft/client/sounds/SoundEngine.mapping:131]:
```suggestion
			COMMENT the SoundInstance to check
```

File: data/net/minecraft/client/sounds/Weighted.mapping [hunk 31]
```diff
@@ -1,3 +1,19 @@
...
+		COMMENT @return The sound associated with the element
+		ARG 1 randomSource
+			COMMENT The random source used for sound selection
+	METHOD getWeight ()I
+		COMMENT {@return The weight of the element}
 	METHOD preloadIfRequired (Lnet/minecraft/client/sounds/SoundEngine;)V
+		COMMENT Preloads the sound if required by the sound engine.
+		COMMENT This method is called to preload the sound associated with the element into the sound engine, ensuring it is ready for playback.
 		ARG 1 engine
+			COMMENT The sound engine used for sound preloading
```
Comment (Matyrobbrt) [inline data/net/minecraft/client/sounds/Weighted.mapping]:
```suggestion
			COMMENT the sound engine used for sound preloading
```

File: data/net/minecraft/client/sounds/Weighted.mapping [hunk 32]
```diff
@@ -1,3 +1,19 @@
 CLASS net/minecraft/client/sounds/Weighted
+	COMMENT The Weighted interface represents an element with a weight in a weighted collection.
+	COMMENT It is used to provide weighted selection and retrieval of elements.
+	COMMENT
+	COMMENT @param <T> The type of the element
```
Comment (Matyrobbrt) [inline data/net/minecraft/client/sounds/Weighted.mapping:5]:
```suggestion
	COMMENT @param <T> the type of the element
```

**Review by Matyrobbrt — CHANGES_REQUESTED** [review]

File: data/net/minecraft/client/sounds/SoundEngine.mapping (hunk 24 above)
Comment (Lanse505) [reply data/net/minecraft/client/sounds/SoundEngine.mapping]: Please rewrite it in Passive Voice :thinkies:

**Review by Matyrobbrt — APPROVED** [review]
:shipit:

**Pull request merged**